src/hidimstat/marginal/partial_dependence_plot.py

```diff
@@ -0,0 +1,807 @@
+            except TypeError as exc:
+                # `np.unique` will fail in the presence of `np.nan` and `str` categories
+                # due to sorting. Temporary, we reraise an error explaining the problem.
+                raise ValueError(
+                    f"The column #{feature_idx} contains mixed data types. Finding unique "
+                    "categories fail due to sorting. It usually means that the column "
+                    "contains `np.nan` values together with `str` categories. Such use "
+                    "case is not yet supported in scikit-learn."
+                ) from exc
+
+            if is_cat or uniques.shape[0] < grid_resolution:
+                # Use the unique values either because:
+                # - feature has low resolution use unique values
+                # - feature is categorical
+                axis = uniques
+            else:
+                # create axis based on percentiles and grid resolution
+                if resolution_statistique:
+                    axis = np.unique(
+                        mquantiles(
+                            data,
+                            prob=np.linspace(0.0, 1.0, grid_resolution + 1),
+                            axis=0,
+                        )
+                    )
+                else:
+                    emp_percentiles = mquantiles(data, prob=percentiles, axis=0)
+                    if np.allclose(emp_percentiles[0], emp_percentiles[1]):
+                        raise ValueError(
+                            "percentiles are too close to each other, "
+                            "unable to build the grid. Please choose percentiles "
+                            "that are further apart."
+                        )
+                    axis = np.linspace(
+                        emp_percentiles[0],
+                        emp_percentiles[1],
+                        num=grid_resolution,
+                        endpoint=True,
+                    )
+        values.append(axis)
+        if not is_cat:
+            # -1 is for correction of the number of classes
+            digitize = np.digitize(data, axis, right=True) - 1
+        else:
+            # convert the object in string if it's necesarry
+            string_to_num = {s: i for i, s in enumerate(sorted(set(data)))}
+            digitize = np.digitize(
+                np.array([string_to_num[s] for s in data]),
+                [string_to_num[s] for s in axis],
+                right=True,
+            )
+        indexes.append(np.clip(digitize, 0, None))
+    return values, indexes
+
+
+def _partial_dependence_brute(est, grid, features, X, method, n_jobs):
+    """
+    Calculate partial dependence using the brute force method.
+    For each value in `grid`, replaces the target features with that value for all samples
+    in X, makes predictions, and averages them. This computes the mean model response
+    across the data distribution for each grid point.
+    Parameters
```

Comment:
Please drop a line before for readability

Reply:
done